File: algorithms_java/sort/Gnome.java
/*Source: https://github.com/TheAlgorithms/Java/blob/master/Sorts/GnomeSort.java */
public class Gnome{
	private static boolean less(Comparable v, Comparable w){
		return (v.compareTo(w) < 0);
	}

	private static void exch(Comparable[] a, int i, int j){
		Comparable swap = a[i];
		a[i] = a[j];
		a[j] = swap;
	}

	public static <T extends Comparable<T>> T[] gnomeSort(T[] arr) {
    	int i = 1;
    	int j = 2;
    	while (i < arr.length) {
      		if (less(arr[i - 1], arr[i])) i = j++;
      		else {
        		exch(arr, i - 1, i);
        		if (--i == 0) {
          			i = j++;
        		}
      		}
    	}
    return null;
  }
}

File: algorithms_python/sort/gnome.py
#source: https://github.com/TheAlgorithms/Python/blob/master/sorts/gnome_sort.py
def gnomeSort(lst):
	if len(lst) <= 1:
		return lst

	i = 1

	while i < len(lst):
		if lst[i - 1] <= lst[i]:
			i += 1
		else:
			lst[i - 1], lst[i] = lst[i], lst[i - 1]
			i -= 1
			if i == 0:
				i = 1

	return lst
